### ISSUE TITLE
Poise::Error Unable to find cookbook for file

### ISSUE THREAD

**Issue by isbaran** [johnbellone/confd-cookbook#1]
Hello, 
Not sure if this is an issue or a configuration problem on my side, but without any tweaks to the cookbook, i get that error:

https://gist.github.com/isbaran/61cc7a7d4b2ec4c4035d

Looks like you pasted the same error in May, what is the fix for that and can you please update the readme if it is a configuration problem ?

Thanks

**Comment by johnbellone** [comment]
What's your wrapper cookbook doing?

**Comment by isbaran** [comment]
Nothing, i get the same error if i add confd cookbook and configuration options to run_list, like:

default['confd']['config'] = {
    'backend'  => 'etcd',
    'interval' => 10,
    'nodes'    => ["xyz"]
}

Chef: 12.3.0

**Comment by johnbellone** [comment]
Are you running chef-client directly on this node? For debugging please set the node attribute POISE_DEBUG=true and run chef-client with the -ldebug command. So something like this:
```ruby
node.default['POISE_DEBUG'] = true
```

And execute chef-client:
```sh
$ chef-client -l debug
```

**Comment by isbaran** [comment]
Yes, i'm running chef-client directly on the node, updated the gist with debug information.

Only change to the cookbook are those lines:
+++ b/cookbooks/poise/files/halite_gem/poise/utils.rb
       Chef::Log.debug("[Poise] Checking cookbook for #{filename.inspect}")
+      Chef::Log.debug("[Poise] run_context.cookbook_collection: #{run_context.cookbook_collection}")
+      Chef::Log.debug("[Poise] run_context: #{run_context}")
+      Chef::Log.debug("[Poise] run_context.node: #{run_context.node}")

# /opt/chef/embedded/bin/ruby --version
ruby 2.1.4p265 (2014-10-27 revision 48166) [x86_64-linux]
# /opt/chef/embedded/bin/gem --version
2.4.4
# /opt/chef/bin/chef-client --version
Chef: 12.3.0

**Comment by tduzan** [comment]
I'm having a similar issue with a different cookbook, this issue is the only reference to "cookbook_collection empty" searching on Google.  Were you able to identify the cause of this and resolve it?  If so, I'd be interested to know how.

**Comment by johnbellone** [comment]
I would be interesting in knowing how I can replicate this.
